libs/dal/src/repositories/notification-template/notification-template.schema.ts

```diff
@@ -248,6 +248,14 @@ notificationTemplateSchema.virtual('notificationGroup', {
 notificationTemplateSchema.index({
   _organizationId: 1,
   'triggers.identifier': 1,
+  deleted: 1,
+});
+
+notificationTemplateSchema.index({
+  _environmentId: 1,
+  _organizationId: 1,
+  createdAt: -1,
+  deleted: 1,
```

Comment:
It seems that Pawel plans to create a new index in this collection. I think it would be better to combine the work if possible to minimize workload for both us and the database.

It's worth noting that including organization in this index may not necessarily optimize the query performance. Since the query primarily filters by environment first, sorting by the organization might not add significant value, in addition, we need to consider that if the organization is missing in the query, other parameters like createdAt and deleted won't be utilized in the query by MongoDB.